main.tf

```diff
@@ -6,16 +6,16 @@ data "aws_eks_cluster" "this" {
   name = local.cluster_id
 }
 
-# Equivalent of aws ecr get-login
-data "aws_ecrpublic_authorization_token" "token" {}
-
 locals {
   # this makes downstream resources wait for data plane to be ready
   cluster_id = time_sleep.dataplane.triggers["cluster_id"]
   region     = data.aws_region.current.name
 
   eks_oidc_issuer_url = replace(data.aws_eks_cluster.this.identity[0].oidc[0].issuer, "https://", "")
 
+  ecrpublic_username = var.ecrpublic_username
```

Comment:
nit: We don't need to re-wrap these as locals. Just use them directly. You can delete these two lines here and just change the `local.ecr*` to `var.ecr*`